### PR TITLE
force public enums to be 64bit wide

### PR DESCRIPTION
**Pull request by poettering**
another attempt to making these enums stable for future extension

### CODE REVIEW

**Comment by floppym** [comment]
I'm curious why you would need 64-bits for a typical enum. Do you anticipate having more than 2 billion distinct values?

I think this really only makes sense for enums that define individuals bits used in bitfields.

**Comment by vcaputo** [comment]
Is the thinking that maybe at some future point the higher, unused bits in the enum may become repurposed as a bitmask?

**Review by bluca — APPROVED** [review]
I like this approach, very nice and clean! LGTM

**Comment by poettering** [comment]
> I'm curious why you would need 64-bits for a typical enum. Do you anticipate having more than 2 billion distinct values?

nope, i do not. however, you never know... in src/resolve/resolved-dns-scope.h we have an enum that started out with three values: yes, no, maybe (its an enum that encodes whether a domain name matches the search/routing domains of a certain interface). But eventually this had to be upgraded because we needed many levels of "yes": if a hostname matches a search domain of 5 labels (e.g. search domain `foo.bar.waldo.foocorp.com`) that's a lot more yes on the yes scale than if it matches another interface's 3 label search domain (e.g. search domain `waldo.foocorp.com`). And suddenly we needed 128 levels of "yes", because there can be up to 127 labels in a domain name at max.

Things like that happen. IRL. in our codebase. i.e. enums that have a _BASE value or multiple that then is used in some "additive" form, and where hence maybe not 2 billion distinct named enum values exist, but a few plus a some bases plus some bases with some offset added on them. 

(moreover we have some enums we started to OR flags into. if you have 64bit you can do that quite safely without risking collision with the non-flag part of the enum)

also: today's CPUs more often than not have 64bit word width, hence when operating with these enum types it's not really that much of a benefit to use less...

**Comment by poettering** [comment]
@keszybz your take on this solution to the problem please!

File: src/shared/wifi-util.c [hunk 1]
```diff
@@ -42,7 +42,7 @@ int wifi_get_interface(sd_netlink *genl, int ifindex, enum nl80211_iftype *iftyp
         if (r < 0)
                 return log_debug_errno(r, "Failed to determine genl family: %m");
         if (family != SD_GENL_NL80211) {
-                log_debug("Received message of unexpected genl family %u, ignoring.", family);
+                log_debug("Received message of unexpected genl family %" PRIi64 ", ignoring.", family);
```
Comment (vcaputo) [inline src/shared/wifi-util.c:45]: why PRIi64 here and PRIu64 below?

Comment (poettering) [reply src/shared/wifi-util.c:45]: no reason, just lennart making a typo. will fix shortly.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**